### PR TITLE
Make ProdVaultTest configurable from the command line

### PR DESCRIPTION
**Pull request by prevostc**
Same as `ProdVaultTest` but:
- an addressbook util to fetch data from the `blockchain-addressbook` package
- a network util to dynamically use a hardhat fork as a network
- moved from DsTest to forge Standard lib

Usage:
```
CHAIN=ethereum VAULT=0x047c41817954b51309a2bd6f60e47bC115C23f1F forge test --match-contract ChainVaultsTest --ffi

CHAIN=fantom VAULT=0xf710D90A1Bd702dAEa619eEbbe876e7085c2A1df forge test --match-contract ChainVaultsTest --ffi
```

### CODE REVIEW

File: forge/test/vault/ChainVaultsTest.t.sol [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    }
+
+    function test_harvestOnDeposit() external {
+        bool harvestOnDeposit = strategy.harvestOnDeposit();
+        if (harvestOnDeposit) {
+            console.log("Vault is harvestOnDeposit.");
+            assertEq(strategy.withdrawalFee(), 0, "Vault is harvestOnDeposit but has withdrawal fee.");
+        } else {
+            console.log("Vault is NOT harvestOnDeposit.");
+            assertEq(strategy.keeper(), addressBookBeefy.keeper, "Vault is not harvestOnDeposit but doesn't have withdrawal fee.");
```
Comment (roman-monk) [inline forge/test/vault/ChainVaultsTest.t.sol:194]: this does not check withdrawal fees 😄 but i figured it is copied from existing ProdTest

**Review by roman-monk — APPROVED** [review]
super cool stuff!

would it be possible to extend ChainVaultsTest to test some specific vaults?
i guess it should work fine and ChainVaultsTest can still be run as independent.
```
SpecificVaultTest is ChainVaultsTest {
  setUp() {
     super.setUp();
  }
}
```

does it make sense to have an option to fork from specified block number?
something like:
```
uint blockNum = vm.envOr("BLOCK", uint(0));
if (blockNum > 0) net.createHardhatNetworkFork(chain, blockNum);
else net.createHardhatNetworkFork(chain);
``` 
this way it will be cached and subsequent tests will run in just milliseconds.
very handy if need to test several just deployed vaults.

**Comment by prevostc** [comment]
Thx for looking at this sir @roman-monk 🙏 
To be honnest, I didn't think this one too much, I was just too lazy to go fetch the right addressbook keeper and owners addresses  each time I wanted to use `ProdVaultTest` :D 

> would it be possible to extend ChainVaultsTest to test some specific vaults?
> i guess it should work fine and ChainVaultsTest can still be run as independent.

I agree, ChainVaultsTest could be built on either in solidity or in a simple bash script. Not sure what's best though.

> does it make sense to have an option to fork from specified block number?
> this way it will be cached and subsequent tests will run in just milliseconds.
> very handy if need to test several just deployed vaults.

I'm not sure the forge fork data is persistent accross each exec of forge. So testing multiple vaults would probably still fetch the fork data on each execution even if the block number is specified.

But it's possible to easily build something like by doing either:
- create a new test dedicated to this use case. Should be easy using the new `HardhatNetworkManager` and `AddressBook` utils
- update `ADDRESS` to support multiple addresses, each one will run the same fork so they would be from the same block/cache
- Add an option to skip the fork part in solidity use a regular avil fork, then use the `--fork-url` option of forge

I'm not sure which is best though, probably the second solution is easier but the output might be difficult to analyze.

**Comment by roman-monk** [comment]
> I'm not sure the forge fork data is persistent accross each exec of forge. So testing multiple vaults would probably still fetch the fork data on each execution even if the block number is specified.

yeah its persistent, im using it in convex tests

**Comment by prevostc** [comment]
Oh nice, let me add that real quick then

**Comment by prevostc** [comment]
You're right, goes from ~15s to ~5s on subsequent runs when specifying block number 👏 
https://github.com/beefyfinance/beefy-contracts/pull/234

**Pull request merged**